src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,260 @@
+                genotypeAlleles.set(genotypeAlleles.indexOf(oldAllele), newAllele);
+                gb.alleles(genotypeAlleles);
+                newGTs.add(gb.make());
+            }
+        }
+        vcBuilder.genotypes(newGTs);
+        return vcBuilder.make();
+    }
+
+    /**
+     * Create the new GenotypesContext with the subsetted PLs and ADs
+     * <p>
+     * Will reorder subsetted alleles according to the ordering provided by the list allelesToKeep
+     *
+     * @param originalGs      the original GenotypesContext
+     * @param originalAlleles the original alleles
+     * @param allelesToKeep   the subset of alleles to use with the new Genotypes
+     * @return a new non-null GenotypesContext
+     */
+    public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
+                                                 final List<Allele> originalAlleles,
+                                                 final List<Allele> allelesToKeep) {
+        nonNull(originalGs, "original GenotypesContext must not be null");
+        nonNull(allelesToKeep, "allelesToKeep is null");
+        nonEmpty(allelesToKeep, "must keep at least one allele");
+        validateTrue(allelesToKeep.get(0).isReference(), "First allele must be the reference allele");
+        validateTrue(allelesToKeep.stream().allMatch(originalAlleles::contains), "OriginalAlleles must contain allelesToKeep");
+
+        int indexOfLast = -1;
+        for (Allele a : allelesToKeep) {
+            validateTrue(indexOfLast < originalAlleles.indexOf(a), "alleles to keep must maintain the order of the original alleles");
+            indexOfLast = originalAlleles.indexOf(a);
+        }
+
+        final int allelesIndex[] = allelesToKeep.stream().mapToInt(originalAlleles::indexOf).toArray();
+        final GenotypesContext newGTs = GenotypesContext.create(originalGs.size());
+        int[] subsettedLikelihoodIndices = subsettedPLIndices(originalAlleles, allelesToKeep);
+
+        for (final Genotype g : originalGs) {
+            validateTrue(g.getPloidy() == 2, "only implemented for ploidy 2 for now");
+
+            final int expectedNumLikelihoods = GenotypeLikelihoods.numLikelihoods(allelesToKeep.size(), 2);
+            // create the new likelihoods array from the alleles we are to use
+            int[] newPLs = null;
+            double newLog10GQ = -1;
+            if (g.hasLikelihoods()) {
+
+                int[] originalPLs = g.getPL();
+
+                if (originalPLs.length != expectedNumLikelihoods) {
+                    newPLs = Arrays.stream(subsettedLikelihoodIndices)
+                            .map(idx -> originalPLs[idx]).toArray();
+                    final int minLikelihood = MathUtil.min(newPLs);
+                    for (int i = 0; i < expectedNumLikelihoods; i++) {
+                        newPLs[i] = newPLs[i] - minLikelihood;
+                    }
+                    final int PLindex = MathUtil.indexOfMin(newPLs);
```

Comment:
can we rename this was clearer name ( that doesn't start with a capital letter...) maybe `indexOfBestGenotype` or something along those lines?

Reply:
done